config.go

```diff
@@ -0,0 +1,119 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/knadh/koanf"
+	"github.com/knadh/koanf/parsers/yaml"
+	"github.com/knadh/koanf/providers/env"
+	"github.com/knadh/koanf/providers/file"
+	"github.com/knadh/koanf/providers/posflag"
+	flag "github.com/spf13/pflag"
+)
+
+type configuration struct {
+	RefreshInterval int                 `koanf:"interval"`
+	Secret          secretConfiguration `koanf:"secret"`
+	Oidc            oidcConfiguration   `koanf:"oidc"`
+	Log             logConfiguration    `koanf:"log"`
+
+	DummyProvider bool `koanf:"dummyprovider"`
+}
+
+type secretConfiguration struct {
+	Name      string `koanf:"name"`
+	Namespace string `koanf:"namespace"`
+	Key       string `koanf:"key"`
```

Comment:
just FYI if the name is the same as the struct property name, the `koanf` tag is not strictly necessary, but no hard feeling if you want to keep it explicit

Reply:
I wasn't sure if it kept the uppercase first letter, so I decided to just be more explicit.